python/src/trezorlib/ethereum.py

```diff
@@ -141,24 +157,181 @@ def encode_data(value: Any, type_name: str) -> bytes:
     raise ValueError(f"Unsupported data type for direct field encoding: {type_name}")
 
 
+def download_from_url(url: str) -> bytes:
+    try:
+        r = requests.get(url)
+        r.raise_for_status()
+        return r.content
+    except requests.exceptions.HTTPError as err:
+        raise RuntimeError(err.response)
+
+
+def get_network_definition_url(
+    chain_id: Optional[int] = None, slip44: Optional[int] = None
+) -> str:
+    if not ((chain_id is None) != (slip44 is None)):  # not XOR
```

Comment:
Could have a wrapper dataclass (similarly as mentioned for `EthDefinitionGetter`) for validating and wrapping these two values so that only one can ever be defined

This is used multiple times in this file, so it could be worth (getting rid of the `ValueError` duplications)

Reply:
code was removed